java-sdk/README.asciidoc

```diff
@@ -0,0 +1,212 @@
+= Lucidworks Connector Java SDK
+
+== Overview
+The Lucidworks Java Connector SDK provides a set of components to support building all types of Fusion connectors.
+Connectors can be created by implementing some of the classes described in this document and using the provided build tool.
+
+Connectors can be deployed directly within a Fusion cluster or outside Fusion as a remote client.
+To deploy a connector into Fusion, upload the build artifact (as a zip file) to the Fusion blob API.
+To deploy remotely, you can use the 
+ifdef::env-github[]
+link:plugin-client.asciidoc[Connector client utility].
+endif::[]
+ifndef::env-github[]
+link:/fusion-server/{version}/search-development/getting-data-in/connectors/sdk/java-sdk/java-connector-dev.html#plugin-client[Connector client utility].
+endif::[]
+
+=== Examples
+The example connectors are a great place to start understanding how to build a Java SDK Connector.
+Have a look at the https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/README.md[example connectors documentation^].
+
+=== Security
+Read more on how to set up a secure environment for Java SDK connectors 
+ifdef::env-github[]
+link:security.asciidoc[here].
+endif::[]
+ifndef::env-github[]
+link:/fusion-server/{version}/search-development/getting-data-in/connectors/sdk/java-sdk/java-connector-dev.html#security[here].
+endif::[]
+
+== Java SDK Framework Overview
+Only a few components must be implemented to compile and build a functioning plugin.
+Here, we give an overview of each of these primary components and related concepts. We start by describing a minimal
+project directory layout, followed by details on the components.
+
+=== Project Layout
+This documentation and examples use Gradle for building connectors. Other build tools can be used instead.
+Check out the https://docs.gradle.org/current/userguide/tutorial_java_projects.html[Gradle quickstart^] for Java Projects.
+
+The minimal project layout is as follows:
+
+    my-connector
+    ├── build.gradle
+    ├── gradle.properties
+    ├── settings.gradle
+    └── src
+        └── main
+            └── java
+                └── Plugin.java
+                └── Config.java
+                └── Fetcher.java
+
+==== build.gradle
+This is a typical Gradle build file. A few important settings are required. You can have a look
+at an example https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/build.gradle[here^].
```

Comment:
Can these URLs be relative so that they match the active branch? Otherwise if someone is viewing say, the 4.1.0 branch and they click these links, they'll go to "master" (develop).

Reply:
Relative or dynamically linked to the current branch?

Reply:
I fixed this by making the link conditional on whether the file is viewed in Github or not.  If viewed in Github, it's a relative link.  If not, it's a FQURL that points to master, for docs site purposes.